app/src/main/java/com/github/wnder/picture/FirebasePicturesDatabase.java

```diff
@@ -121,7 +120,7 @@ public FirebasePicturesDatabase(){
     }
 
     @Override
-    public CompletableFuture<Void> sendUserGuess(String uniqueId, String user, Location guessedLocation) {
+    public CompletableFuture<Void> sendUserGuess(String uniqueId, String user, Location guessedLocation, Bitmap mapSnapshot) {
```

Comment:
Maybe we could store the mapSnapshot on the remote db, so that even when the picture is not stored locally we can retrieve it and have access to its page.

Reply:
At least I think that you should prevent the app from crashing when you open a picture in your history that isn't locally stored, addressing Romain's issue higher should do the job though

Reply:
I went with another approach: if there is no map snapshot in the local database but we can get the necessary metadata, it is possible to generate one on the flight. In case no internet connection is available it no longer crashes but shows a blank activity for now (could be replaced by an alert of a default "no map" picture in a subsequent PR).

Reply:
That's actually a nice solution!